File: Task2/apz-pzpi-21-11-milennyi-ivan-task2/src/view_models/shearing_log_vm.rs
use serde::Serialize;
use sqlx::FromRow;
use utoipa::ToSchema;

#[derive(Debug, Serialize, FromRow, ToSchema)]
pub(crate) struct ShearingLogVM{
    id: u64,
    timestamp: u64,
    wool_amount: u64,
    shepherd_name: Option<String>,
    shepherd_surname: Option<String>,
    sheep_name: String
}

File: Task4/apz-pzpi-21-11-milennyi-ivan-task4/src/viewModels/ShearingLogVM.ts
import { timestampToDate } from "../utils/helpers";

export class ShearingLogVM {
    id: number;
    date: string;
    woolAmount: number;
    shepherdName: string | null;
    shepherdSurname: string | null;
    sheepId: number;

    constructor(id: number, timestamp: number, woolAmount: number, sheepId: number, shepherdName: string | null, shepherdSurname: string | null) {
        this.id = id;
        this.woolAmount = woolAmount / 1000;
        this.shepherdName = shepherdName;
        this.shepherdSurname = shepherdSurname;
        this.sheepId = sheepId;
        this.date = timestampToDate(timestamp);
    }
}